rockcraft/project.py

```diff
@@ -428,6 +429,24 @@ def _validate_package_repositories(
 
         return package_repositories
 
+    def to_yaml(self) -> str:
```

Comment:
I changed this from `yaml()` to `to_yaml()` because mypy was complaining about this being an "invalid" override, since `YamlModel.yaml()` takes many parameters and has typing issues of its own (like boolean parameters with a `None` default).

Reply:
(Out of scope here, but a general thought:) Do we want to make this a standard thing for *craft projects? We could pop a `CraftYamlModel` into `craft-application` that does this and inherit from that

Reply:
this was the original intention. But YamlModel already has a `yaml` method, might be worth looking into API compatibility a bit more